ctrd/container.go

```diff
@@ -500,3 +500,36 @@ func (c *Client) ResizeContainer(ctx context.Context, id string, opts types.Resi
 
 	return pack.task.Resize(ctx, uint32(opts.Height), uint32(opts.Width))
 }
+
+// WaitContainer waits until container's status is stopped.
+func (c *Client) WaitContainer(ctx context.Context, id string) (types.ContainerWaitOKBody, error) {
+	wrapperCli, err := c.Get(ctx)
+	if err != nil {
+		return types.ContainerWaitOKBody{
+			Error:      err.Error(),
+			StatusCode: -1,
+		}, fmt.Errorf("failed to get a containerd grpc client: %v", err)
+	}
+
+	ctx = leases.WithLease(ctx, wrapperCli.lease.ID())
+
+	waitExit := func() *Message {
+		return c.ProbeContainer(ctx, id, -1*time.Second)
+	}
+
+	var msg *Message
+	// wait for the task to exit.
+	msg = waitExit()
+
+	if err := msg.RawError(); err != nil && errtypes.IsTimeout(err) {
```

Comment:
If the `errtypes.IsTimeout(err)` is true, we can return `types.ContainerWaitOKBody{}, err` like my last comment. WDYT?

Reply:
OK, I'll fix all.

Reply:
Solved

Reply:
emmm... could we remove this duplicate line, since you have done it in the following statement?